spec/support/controller_helpers.rb

```diff
@@ -0,0 +1,12 @@
+module ControllerHelpers
+  def login_with(user = double('user'), scope = :user)
+    current_user = "current_#{scope}".to_sym
+    if user.nil?
+      allow(request.env['warden']).to receive(authenticate!).and_throw(:warden, scope: scope)
+      allow(controller).to receive(current_user).and_return(nil)
+    else
+      allow(request.env['warden']).to receive(:authenticate!).and_return(user)
+      allow(controller).to receive(current_user).and_return(user)
+    end
+  end
+end
```

Comment:
We had a chat about this. The authenticate method is being used (which is how devise is supposed to be use) which requires the warden stubbing. We don't always use devise in this way which is why we regularly dont have to stub warden.